### PR TITLE
Docs lint/test traning data

### PR DESCRIPTION
**Pull request by m-vdb**
**Proposed changes**:
- Fix https://github.com/RasaHQ/rasa/issues/6399
- Tech writer API: use `yaml-rasa` or `yml-rasa` code blocks in docs
- Logic: created a simple python test that loads every docs file, parses it for codeblocks and validates the training data using `RasaYAMLReader`

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by erohmensing** [comment]
bless you for working on this 🙏 i was looking into tagging all of them this morning -- happy to put in the manual work of doing that once this is ready

**Comment by m-vdb** [comment]
@erohmensing it's ready! If you could flag all the training data that we want to validate in the docs, that'd be awesome. I've added a description to the PR so that it's easier to understand, but anyway it's not super hard to use. I've already added a CI step, and you can run the checks locally with `make test-docs`. It's highly probable that some use cases are not supported yet - bear with me!

**Comment by m-vdb** [comment]
@erohmensing yes sure, will do 👍

**Comment by erohmensing** [comment]
Otherwise I believe we are good! It can't seem to get some files for some reason, not sure what's up with that

**Comment by m-vdb** [comment]
> It can't seem to get some files for some reason

@erohmensing which files?

**Comment by erohmensing** [comment]
Oh wait, maybe someone updated them without updating them in the docs, looks like the autoconfig ones https://github.com/RasaHQ/rasa/runs/1014039899#step:13:113

**Comment by m-vdb** [comment]
@erohmensing no it's a problem that has been introduced with this PR. Found the root cause, on it

File: .github/workflows/continous-integration.yml [hunk 1]
```diff
@@ -101,11 +101,14 @@ jobs:
       # trying to run the `rasa` command once before the docs build.
       run: poetry run rasa --help
 
-    - name: Test Docs 📃
+    - name: Build Docs 📃
       run: |
         poetry run python -c "from scripts import release; release.generate_changelog('major.minor.patch')"
```
Comment (erohmensing) [inline .github/workflows/continous-integration.yml:106]: wdyt about adding this directly to `make docs` so its easier to `make docs test-docs` locally?

Comment (m-vdb) [reply .github/workflows/continous-integration.yml:106]: no we can't do that, it's used to generate the docs in the deployment pipeline

Comment (m-vdb) [reply .github/workflows/continous-integration.yml:106]: but we could definitely modify test-docs to do that automatically

Comment (erohmensing) [reply .github/workflows/continous-integration.yml:106]: ah makes sense 👍

Comment (erohmensing) [reply .github/workflows/continous-integration.yml:106]: wait, doesn't the changelog still need to be generated though 🤔

Comment (m-vdb) [reply .github/workflows/continous-integration.yml:106]: yes but it's not done like this. Look at the "major.minor.patch" argument :)

Comment (m-vdb) [reply .github/workflows/continous-integration.yml]: I updated the workflow here: c55905a (and added a helper). It should be easier to use locally now 👍

**Review by erohmensing — APPROVED** [review]
🥳

**Pull request merged**